### ISSUE TITLE
Column selection - cannot paste with virtual whitespace

### ISSUE THREAD

**Issue by ArashPartow** [microsoft/vscode#43145]
I have the following bit of data

```
col1   col2
aaa    bbbbbbbbb
cccccc ddddd
ee     fffffffffffff
aaa    bbbbbbbbb
cccccc ddddd
ee     ffffffffffffffffff
```

I would like to be able to select the 2nd column (aka col2) and paste it next to the original col2. 

This operation is quite trivial in editors such as _Visual Studio, UltraEdit, NotePad++, QtCreator, emacs etc.._  The following shows UltraEdit, NotePad++, Visual Studio, QtCreator and Kate column select modes:

![vscode_column_select_problem01](https://user-images.githubusercontent.com/513642/35946358-92176174-0cb7-11e8-9eda-2faa6ed22c64.png)

![msvc](https://user-images.githubusercontent.com/513642/39613166-ba3d0058-4fa8-11e8-8466-e5830f33671a.png)

![qtcreator](https://user-images.githubusercontent.com/513642/39677997-d1612b08-51c7-11e8-98af-f2f495d3d6a8.png)

![image](https://user-images.githubusercontent.com/513642/44886538-1e093c00-ad0a-11e8-8ccb-18719c713b66.png)

----

However when I try to do it using vscode, I encounter a very _**non-intuitive**_ outcome:

![vscode_column_select_problem02](https://user-images.githubusercontent.com/513642/35946383-ae44f546-0cb7-11e8-8a12-60e43a913e83.png)

Is there a plugin or correct way to select a column of text and paste it as-is into a text? _(or the same one)_

----

@alexandrudima the  **Problems** and **Issues** observed:

1. Holes appear in the selected rectangular area
2. Rows are selected at different lengths instead of max number of columns
3. When pasting what has been selected, the result is non-intuitive
4. Behaviour differs when using keyboard only versus keyboard-mouse combination

**Comment by phiresky** [comment]
Note that you kan _kind of_ do this by using column selection (shift+alt drag)

![image](https://user-images.githubusercontent.com/2303841/39622359-3637e9aa-4f92-11e8-95be-19241694c145.png)

then pasting in multi cursor mode 
![image](https://user-images.githubusercontent.com/2303841/39622391-5171b5fc-4f92-11e8-9950-5a64a2e48a88.png)

**Comment by ArashPartow** [comment]
@phiresky  please review the following section in the issue post: **_"the Problems and Issues observed:"_**

Found at the bottom of the issue report - specifically **Point 3**

-----

Note there are numerous problems, such as the editor allowing the user to get into non-intuitive/ridiculous selection modes, not being able to properly align the pasted text, overlapping selects results in the previously selected text to become unselected (holes et al) and many other issues.

**Comment by phiresky** [comment]
1. no holes appear for me for your sample as shown in the above screenshots
2. rows are selected as a block / as expected for me, except when the line length is shorter than the selected block, then it is selected to the end of the line
3. I don't understand your last  screenshot or what you mean by non-intuitive. 
4. not sure what combination you mean since you don't specify what commands you are using

"non-intuitive/ridiculous selection modes" is obviously subjective and thus not a helpful assertion.
"many other issues" isn't really 

pasting N lines in vscode causes them to be spread across the active cursors if the number of cursors is also N.

**Comment by ArashPartow** [comment]
@phiresky thank-you for your comments, lets allow the Microsoft guys to resolve this issue.

**Comment by gazzar** [comment]
This is basically a dupe of #5940

**Comment by ArashPartow** [comment]
@gazzar: Just to let you know   @alexandrudima  was assigned this task - but he seems to have dropped it.

**Comment by gazzar** [comment]
@ArashPartow Yes, it's mysterious and frustrating. And he was assigned to #73837 a few days ago, so at least one other person in Microsoft is now aware that this fundamental editor feature is still missing from VSCode.

**Comment by alexandrudima** [comment]
Let's track in #5940.

**Comment by sandipchitale** [comment]
This extension may help:

https://marketplace.visualstudio.com/items?itemName=sandipchitale.exchange-selection-start-end

**Comment by NurjanovUlugbek** [comment]
Please make this feature copy past normally column mode like notepad++ ultraedit
